src/token/getTokens.ts

```diff
@@ -0,0 +1,57 @@
+import { ListSwapAssets } from '../definitions/swap';
+import { Token, RawTokenData, GetTokensOptions } from './types';
+import { sendRequest, JSONRPCError } from '../queries/request';
+
+/**
+ * Retrieves a list of tokens on Base.
+ * 
+ * @param {GetTokensOptions} options - An optional object that can be used to filter the list of tokens.
+ *  options.limit - The maximum number of tokens to return (default: 50).
+ *  options.search - A string to search for in the token name or symbol.
+ *  options.page - The page number to return (default: 1).
+ * @returns A promise that resolves to an array of `Token` objects.
+ * @throws Will throw an error if the request to the RPC URL fails.
+ * @example
+import { getTokens } from '@coinbase/onchainkit'
+
+const tokens = await getTokens({ limit: '1', search: 'degen' });
+// [
+//   {
+//     address: '0x4ed4e862860bed51a9570b96d89af5e1b0efefed',
+//     chainId: 8453,
+//     decimals: 18,
+//     image: 'https://d3r81g40ycuhqg.cloudfront.net/wallet/wais/3b/bf/3bbf118b5e6dc2f9e7fc607a6e7526647b4ba8f0bea87125f971446d57b296d2-MDNmNjY0MmEtNGFiZi00N2I0LWIwMTItMDUyMzg2ZDZhMWNm',
+//     name: 'DEGEN',
+//     symbol: 'DEGEN'
+//   }
+// ]
+*/
+export async function getTokens(options?: GetTokensOptions): Promise<Token[] | JSONRPCError> {
+  // Default filter values
+  const defaultFilter: GetTokensOptions = {
+    limit: '50',
+    page: '1',
+  };
+
+  const filters = { ...defaultFilter, ...options };
+
+  try {
+    const res = await sendRequest<GetTokensOptions, RawTokenData[]>(ListSwapAssets, [filters]);
+
+    if (res.error) {
+      return res.error;
```

Comment:
one thing to consider is if we want to throw an error, or return a `JSONRPCError` object

instead of using `try/catch`, users can try to cast the response to this object and handle the error appropriately
```
import { JSONRPCError } from '@onchainkit/queries/request';

const resp = await getTokens({ limit: '1', search: 'degen' });

if (resp as JSONRPCError) {
  console.error('error getting tokens', resp);
} else {
  console.log('tokens:', resp);
}
```

Reply:
this is for the case where the server responds with JSON-RPC, but there is an issue with the request (e.g. unauthenticated, over rate limit, invalid param, etc.)

vs. if the `fetch` call fails due to the server being down, etc. in which it will throw an error instead

Reply:
I don't think we should expose `import { JSONRPCError } from '@onchainkit/queries/request';` even thought might be the most Eng-Correct way, it doesn't feel simple enough.

I feel like we should expose a more error specific type for each API. so something
```
import { GetTokensAPIError } from '@coinbase/onchainkit/token'
```

so in some way even if it's a bit reduant if fills more explicit.

Reply:
how about this?

```
import { getTokens } from '../token/index';

const resp = await getTokens({ limit: '1', search: 'degen' });

if ('error' in resp) {
  console.error('error getting tokens', (resp as GetTokensError).error);
} else {
  console.log(resp);
}
```

```
alec@TJ03GKYH46 onchainkit % npx tsx src/examples/swap.ts
error getting tokens 'unauthorized - invalid key'
alec@TJ03GKYH46 onchainkit % npx tsx src/examples/swap.ts
[
  {
    address: '0x4ed4e862860bed51a9570b96d89af5e1b0efefed',
    chainId: 8453,
    decimals: 18,
    image: 'https://d3r81g40ycuhqg.cloudfront.net/wallet/wais/3b/bf/3bbf118b5e6dc2f9e7fc607a6e7526647b4ba8f0bea87125f971446d57b296d2-MDNmNjY0MmEtNGFiZi00N2I0LWIwMTItMDUyMzg2ZDZhMWNm',
    name: 'DEGEN',
    symbol: 'DEGEN'
  }
]
```

Reply:
https://github.com/microsoft/TypeScript/issues/10485

Reply:
yeah I like this, we can get the Frontend Fellowship share some thoughts on our types, but overall this is better on what we had.